test/run-drun/upgrade-mutable/version1.mo

```diff
@@ -0,0 +1,20 @@
+import Prim "mo:prim";
+
+actor {
+   type Entry = { key : Nat; value : Text };
+   type SubEntry = { key : Nat; };
+   
+   stable var array1 : [var Entry] = [var { key = 1; value = "1" }];
+   stable var array2 : [var SubEntry] = [var];
```

Comment:
good test! upgrade from version0 to version1 should not be allowed because of invariance.

Reply:
Thanks a lot!